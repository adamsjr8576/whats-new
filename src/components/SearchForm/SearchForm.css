header {
  width: 100%;
  height: 10vh;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

span {
  color: #20AF99;
}

input {
  width: 32%;
  padding: 10px;
  outline: 0;
  font-size: 135%;
  border: 0;
}

.search-button {
  width: 25%;
  padding: 10px;
  font-size: 135%;
  border-radius: 10px;
  background-color: #B8DCEF;
  cursor: pointer;
  outline: 0;
}

.search-button:hover {
  background-color: #A6C6D8;
}

.search-button:active {
  transform: scale(.9);
}
