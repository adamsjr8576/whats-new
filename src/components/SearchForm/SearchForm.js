import React, {Component} from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      search: ''
    }
  }

  changeInput = event => {
    this.setState({search: event.target.value})
  }

  clearInput = () => {
    this.setState({search: ''});
  }

  searchHandler = () => {
    const { search } = this.state;
    const { articleData, handleSearch } = this.props;

    const filteredArticles = articleData.filter(article => {
      const articleWords = article.headline.split(' ');
      return articleWords.includes(search);
    });
    if (filteredArticles.length > 0) {
      handleSearch(filteredArticles);
    }
    this.clearInput();
  }

  render() {
    const { search } = this.state;
    const { loading } = this.props;

    return (
      <header>
        <h1>What's <span>New?</span></h1>
        <input
          type="text"
          placeholder="Search for news article here"
          name="search"
          value={search}
          onChange={event => this.changeInput(event)}
          onKeyDown={(e) => {
            if (e.key ==="Enter") {
              this.searchHandler();
            }
          }}
          disabled={loading}
        />
        <button className="search-button" disabled={loading} onClick={() => this.searchHandler()}>Search Now</button>
      </header>
    );
  }
}

export default SearchForm;
