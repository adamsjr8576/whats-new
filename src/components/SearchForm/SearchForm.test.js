import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<SearchForm
      handleSearch={jest.fn()}
      articleData={[{
        id: 1,
        headline: 'Spider-Man Will Remain in the Marvel Cinematic Universe',
        img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
        description: 'The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios.',
        url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'
      }]}
      loading={false}
      />)

      expect(wrapper).toMatchSnapshot()
  });

  it('should update state with a new search when changeInput is called', () => {
    const wrapper = shallow(<SearchForm
      handleSearch={jest.fn()}
      articleData={[{
        id: 1,
        headline: 'Spider-Man Will Remain in the Marvel Cinematic Universe',
        img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
        description: 'The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios.',
        url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'
      }]}
      loading={false}
      />);
    const mockEvent = {target: {value: "Denver"}};
    const expected = "Denver";

    wrapper.instance().changeInput(mockEvent);
    expect(wrapper.state('search')).toEqual(expected);
  });

  it('should update state with a new search when clearInput is called', () => {
    const wrapper = shallow(<SearchForm
      handleSearch={jest.fn()}
      articleData={[{
        id: 1,
        headline: 'Spider-Man Will Remain in the Marvel Cinematic Universe',
        img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
        description: 'The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios.',
        url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'
      }]}
      loading={false}
      />);
    const expected = '';

    wrapper.setState({search: "Hello world!!!"});
    wrapper.instance().clearInput();
    expect(wrapper.state("search")).toEqual(expected);
  });

  it('should call clearInput and handleSearch when searchHandler is called', () => {
    const mockHandleSearch = jest.fn()
    const wrapper = shallow(<SearchForm
      handleSearch={mockHandleSearch}
      articleData={[{
        id: 1,
        headline: 'India\'s Lost Moon Lander Is Somewhere in This Photo',
        img: 'https://cdn.mos.cms.futurecdn.net/FsUeVGmY89BWeb36U8cUj5-650-80.jpg',
        description: 'The image was captured by NASA\'s Lunar Reconnaissance Orbiter (LRO) on Sept. 17 as it flew over the targeted landing site of India\'s Chandrayaan-2 mission.',
        url: 'https://www.livescience.com/india-moon-landing-site-nasa-photo.html'
      },
      {
        id: 2,
        headline: 'NASA, in Reddit AMA, says Earth probably won\'t be eaten by a black hole',
        img: 'https://cnet2.cbsistatic.com/img/tM8V3SxstaN-YQiY1QPVMhYBF1k=/2019/09/26/f41e8d77-b008-4887-ae0b-a6dba3e4db60/bh-visualization1.jpg',
        description: 'NASA\'s black hole experts popped up on Reddit on Friday to field questions about these enigmatic objects. ',
        url: 'https://www.cnet.com/news/nasa-in-reddit-ama-says-earth-probably-wont-be-eaten-by-a-black-hole/'
      }]}
      loading={false}
      />);
    const mockfilteredArticles = [
      {
        id: 1,
        headline: 'India\'s Lost Moon Lander Is Somewhere in This Photo',
        img: 'https://cdn.mos.cms.futurecdn.net/FsUeVGmY89BWeb36U8cUj5-650-80.jpg',
        description: 'The image was captured by NASA\'s Lunar Reconnaissance Orbiter (LRO) on Sept. 17 as it flew over the targeted landing site of India\'s Chandrayaan-2 mission.',
        url: 'https://www.livescience.com/india-moon-landing-site-nasa-photo.html'
      }
    ];
    wrapper.setState({search: "Moon"});
    wrapper.instance().clearInput = jest.fn();
    wrapper.instance().searchHandler();

    expect(wrapper.instance().clearInput).toHaveBeenCalled();
    expect(mockHandleSearch).toHaveBeenCalledWith(mockfilteredArticles);
  });

  it('should call the changeInput method with the event when changed', () => {
    const wrapper = shallow(<SearchForm
      handleSearch={jest.fn()}
      articleData={[{
        id: 1,
        headline: 'Spider-Man Will Remain in the Marvel Cinematic Universe',
        img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
        description: 'The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios.',
        url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'
      }]}
      loading={false}
      />);
      wrapper.instance().changeInput = jest.fn();
      wrapper.find('input').simulate('change');
      expect(wrapper.instance().changeInput).toHaveBeenCalled();
  });

  it('should call the searchHandler method when key down is enter', () => {
    const wrapper = shallow(<SearchForm
      handleSearch={jest.fn()}
      articleData={[{
        id: 1,
        headline: 'Spider-Man Will Remain in the Marvel Cinematic Universe',
        img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
        description: 'The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios.',
        url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'
      }]}
      loading={false}
      />);

      wrapper.instance().searchHandler = jest.fn();
      wrapper.find('input').simulate('keyDown', {key: 'Enter'});
      expect(wrapper.instance().searchHandler).toHaveBeenCalled();
  });

  it('should call the searchHandler method when clicked', () => {
    const wrapper = shallow(<SearchForm
      handleSearch={jest.fn()}
      articleData={[{
        id: 1,
        headline: 'Spider-Man Will Remain in the Marvel Cinematic Universe',
        img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
        description: 'The web slinger will be the only superhero with the ability to cross between Disney and Sony’s cinematic universes, according to Marvel Studios.',
        url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'
      }]}
      loading={false}
      />);

      wrapper.instance().searchHandler = jest.fn();
      wrapper.find('button').simulate('click');
      expect(wrapper.instance().searchHandler).toHaveBeenCalled();
  })
});
