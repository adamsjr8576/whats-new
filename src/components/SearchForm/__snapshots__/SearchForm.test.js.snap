// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<header>
  <h1>
    What's 
    <span>
      New?
    </span>
  </h1>
  <input
    disabled={false}
    name="search"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search for news article here"
    type="text"
    value=""
  />
  <button
    className="search-button"
    disabled={false}
    onClick={[Function]}
  >
    Search Now
  </button>
</header>
`;
